lib/phoenix/endpoint/instrument.ex

```diff
@@ -0,0 +1,187 @@
+defmodule Phoenix.Endpoint.Instrument do
+  @moduledoc false
+
+  # This is the arity that event callbacks in the instrumenter modules must
+  # have.
+  @event_callback_arity 3
+
+  def definstrument(otp_app, endpoint) do
+    app_instrumenters = app_instrumenters(otp_app, endpoint)
+
+    quote bind_quoted: [app_instrumenters: app_instrumenters] do
+      @doc """
```

Comment:
Those docs are excellent! We should also include a snippet about instrumentation covering all aspects, from configuring up to defining your own module up to instrument in a new section in `Phoenix.Endpoint`.